src/java.desktop/macosx/classes/sun/java2d/macos/MacOSFlags.java

```diff
@@ -90,19 +95,66 @@ private static boolean getPropertySet(String p) {
     private static void initJavaFlags() {
         java.security.AccessController.doPrivileged(
                 (PrivilegedAction<Object>) () -> {
-                    metalEnabled = getBooleanProp("sun.java2d.metal", false);
-                    if (metalEnabled) {
-                        metalVerbose = isBooleanPropTrueVerbose("sun.java2d.metal");
-
-                        // Check whether Metal framework is available on the system
-                        if (!MTLGraphicsConfig.isMetalAvailable()) {
+                    PropertyState oglState = getBooleanProp("sun.java2d.opengl", PropertyState.UNSPECIFIED);
+                    PropertyState metalState = getBooleanProp("sun.java2d.metal", PropertyState.UNSPECIFIED);
+
+                    if (metalState == PropertyState.UNSPECIFIED) {
+                        if (oglState == PropertyState.DISABLED) {
+                            oglEnabled = false;
+                            metalEnabled = true;
+                        } else if (oglState == PropertyState.ENABLED || oglState == PropertyState.UNSPECIFIED) {
+                            oglEnabled = true;
                             metalEnabled = false;
+                        }
+                    } else if (metalState == PropertyState.ENABLED) {
+                        if (oglState == PropertyState.DISABLED || oglState == PropertyState.UNSPECIFIED) {
+                            oglEnabled = false;
+                            metalEnabled = true;
+                        } else if (oglState == PropertyState.ENABLED) {
+                            oglEnabled = true;
+                            metalEnabled = false;
+                        }
+                    } else if (metalState == PropertyState.DISABLED) {
+                        oglEnabled = true;
+                        metalEnabled = false;
+                    }
+
+                    oglVerbose = isBooleanPropTrueVerbose("sun.java2d.opengl");
+                    metalVerbose = isBooleanPropTrueVerbose("sun.java2d.metal");
+
+                    boolean oglAvailable = CGLGraphicsConfig.isCGLAvailable();
+                    boolean metalAvailable = MTLGraphicsConfig.isMetalAvailable();
```

Comment:
This will attempt to initialize both, loading both native libraries, even in the typical case where we don't fallback. Is this what we want?